WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostPublishSettingsViewModel.kt

```diff
@@ -88,22 +88,28 @@ class EditPostPublishSettingsViewModel @Inject constructor(
             siteStore.getSiteByLocalId(it)
         }
         loadAuthors()
-        viewModelScope.launch {
-            shareLimit = siteModel?.let {
-                getJetpackSocialShareLimitStatusUseCase.execute(it)
-            } ?: ShareLimit.Disabled
-            loadConnections()
-            loadJetpackSocial()
+
+        if (jetpackSocialFeatureConfig.isEnabled()) {
+            viewModelScope.launch {
+                shareLimit = siteModel?.let {
+                    getJetpackSocialShareLimitStatusUseCase.execute(it)
+                } ?: ShareLimit.Disabled
+                loadConnections()
+                loadJetpackSocialIfSupported()
+            }
+        } else {
+            _showJetpackSocialContainer.value = false
         }
     }
 
     fun onScreenShown() {
+        if (!jetpackSocialFeatureConfig.isEnabled()) return
```

Comment:
❓ Would it make sense to move this inside the other `if`,  like this?
```
        if (actionEvents.value is ActionEvent.OpenSocialConnectionsList) {
              if (!jetpackSocialFeatureConfig.isEnabled()) return
              ...
        }
```

The reason I'm asking is because the intention with the `onScreenShown` method was to have something generic that is called basically when `onResume` is called. It's possible that this screen is changed in the future and something has to be done when `onScreenShown` is called that has nothing to do with Jetpack Social. In that case, the developer will have to understand why this feature flag check is being used right at the beginning of the method instead of tied to a feature specific logic (in this case, whether the last action was `OpenSocialConnectionsList` or not).

WDYT?

Reply:
Hey @RenanLukas 👋, 
if I get it right, the goal is keeping the onScreenShown as a generic hook to onResume; if that's the case what if we do something like?

- Propagate the onResume naming into the view model (so rename onScreenShown to onResume that actually keep all the context of the event we are hooking to)
- Condition all the block with an && on the FF like below

```
if (jetpackSocialFeatureConfig.isEnabled() && actionEvents.value is ActionEvent.OpenSocialConnectionsList) { 
...
```

Let me know and I can commit the changes 👍

Reply:
Hey @develric , makes sense to me 👍 thanks!